js/src/signed-varint.js

```diff
@@ -0,0 +1,127 @@
+// @flow
+
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import * as Bytes from './bytes.js';
+
+export const maxVarintLength = 10;
+
+const mathPowTwoThirtyTwo = Math.pow(2, 32);
+
+function toUint32(n: number): number {
+  return n >>> 0;
+}
+
+/**
+ * Encodes `val` as signed varint and writes that into `buf` at `offset`. This returns the number
+ * of bytes written.
+ */
+export function encode(val: number, buf: Uint8Array, offset: number): number {
```

Comment:
The file name already names this signed varint. The reason I renamed it in the importer is to make it clearer in the code since the importer already has encode/decode.